### PR TITLE
Add support for .properties for expand_key mode

### PR DESCRIPTION
**Pull request by kalistace**
.properties files are commonly used among Java based projects. The feature would allow teams to use them out of the box without having to convert their existing properties.

Additionally, it allows application to look for .properties files on the filepath in case consul is down or if we are running an application locally.

The PR brings a new dependency : https://github.com/gagle/node-properties

### CODE REVIEW

**Comment by jsullivan3** [comment]
Thank you for submitting this patch - I won't be able to take an in-depth look at this one for a couple of days, though.  I'll get back to you as soon as I can look through it.

**Comment by jsullivan3** [comment]
The latest commit appears to have broken the build.  Can you please check the value for 'properties on line 24 of package.json?

**Comment by kalistace** [comment]
Sorry, i pushed that on the wrong branch :( 

I'm basically preparing a new pull request of that module allowing tolerant substitution, instead of just storing the file as a flat file when a property cannot be parsed.

As soon as this PR https://github.com/gagle/node-properties/pull/26 is merged, I'll do another PR for that feature.

File: lib/consul/index.js [hunk 1]
```diff
@@ -89,41 +90,94 @@ var populate_kvs_from_json = function(branch, prefix, obj, cb) {
...
+  var handle_as_flat_file = function(fqf, branch, file, cb) {
+    fs.readFile(fqf, {encoding: 'utf8'}, function (err, body) {
       /* istanbul ignore if */
       if (err) return cb('Failed to read key ' + fqf + ' due to ' + err);
       var body = body ? body.trim() : '';
       write_content_to_consul(create_key_name(branch, file), body, cb);
     });
+  };
+
+  var handle_expended_keys_with_different_file_types = function() {
```
Comment (jsullivan3) [inline lib/consul/index.js]: Spelling issue - functionally fine, but I believe this function should be named "_expanded_" and not "_expended_"

File: readme.md [hunk 2]
```diff
@@ -161,7 +163,7 @@ If you would like git2consul to treat JSON documents in your repo as fully forme
 git2consul in expand_keys mode will generate the following KV:
 
 ```
-/expando_keys/root.json/first_level/second_level/third_level/you%20get%20the%20picture
+/expand_keys/root.json/first_level/second_level/third_level/you%20get%20the%20picture
```
Comment (jsullivan3) [inline readme.md]: I think that since the repository is named 'expando_keys' we want to keep the 'o' here, right?

**Comment by jsullivan3** [comment]
Just a couple of very minor changes - looks great.  If you could tweak the spelling issue, I can get this merged.  Thanks for contributing this functionality!

**Comment by kalistace** [comment]
All fixed :) Thanks for reviewing it !

More than happy to contribute.

**Pull request merged**